tests/interface/TWIoTeXStakingTests.cpp

```diff
@@ -0,0 +1,100 @@
+    auto d = parse_hex("07c35fc00102030405060708090a0b0c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003e900000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000e7468697320697320612074657374000000000000000000000000000000000000");
+
+    byte name[12] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+    auto canName = TWDataCreateWithBytes(name, 12);
+    byte t[] = {'t', 'h', 'i', 's', ' ', 'i', 's', ' ', 'a', ' ', 't', 'e', 's', 't'};
+    auto test = TWDataCreateWithBytes(t, 14);
+    auto stake = TWIoTeXStakingStake(canName, 1001, true, test);
+
+    ASSERT_EQ(d, data(TWDataBytes(stake), TWDataSize(stake)));
+}
+
+TEST(TWIoTeXStaking, Unstake) {
+    auto d = parse_hex("c8fd6ed000000000000000000000000000000000000000000000000000000000000003e90000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000e7468697320697320612074657374000000000000000000000000000000000000");
+
+    byte t[] = {'t', 'h', 'i', 's', ' ', 'i', 's', ' ', 'a', ' ', 't', 'e', 's', 't'};
+    auto test = TWDataCreateWithBytes(t, 14);
+    auto unstake = TWIoTeXStakingUnstake(1001, test);
+
+    ASSERT_EQ(d, data(TWDataBytes(unstake), TWDataSize(unstake)));
+}
+
+TEST(TWIoTeXStaking, Withdraw) {
+    auto d = parse_hex("030ba25d00000000000000000000000000000000000000000000000000000000000003e90000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000e7468697320697320612074657374000000000000000000000000000000000000");
+
+    byte t[] = {'t', 'h', 'i', 's', ' ', 'i', 's', ' ', 'a', ' ', 't', 'e', 's', 't'};
+    auto test = TWDataCreateWithBytes(t, 14);
+    auto withdraw = TWIoTeXStakingWithdraw(1001, test);
+
+    ASSERT_EQ(d, data(TWDataBytes(withdraw), TWDataSize(withdraw)));
+}
+
+TEST(TWIoTeXStaking, AddStake) {
+    auto d = parse_hex("6e7b301700000000000000000000000000000000000000000000000000000000000003e900000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000000");
+
+    auto add = TWIoTeXStakingAddStake(1001, nullptr);
+
+    ASSERT_EQ(d, data(TWDataBytes(add), TWDataSize(add)));
+}
+
+TEST(TWIoTeXStaking, MoveStake) {
+    auto d = parse_hex("d3e41fd200000000000000000000000000000000000000000000000000000000000003e90102030405060708090a0b0c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000");
+
+    byte name[12] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+    auto canName = TWDataCreateWithBytes(name, 12);
+    auto move = TWIoTeXStakingMoveStake(1001, canName, nullptr);
+
+    ASSERT_EQ(d, data(TWDataBytes(move), TWDataSize(move)));
+}
+
+TEST(TWIoTeXStaking, Sign) {
```

Comment:
can you add a full test to swift and kotlin?
1. build staking data
2. build signing input
3. sign and check encoded

Reply:
could someone at your side do that? it would probably take 10~20 min, but couple of hours for me to figure out such a test since I am all new to swift and kotlin

Reply:
I can do it if I have time today, can you add some real world staking txs in cpp?